py/sight/demo/multiple_opt_demo.py

```diff
@@ -0,0 +1,332 @@
+        with Attribute("sample_id", id, sight):
+          tasks.append(
+              sight.create_task(
+                  # both base and treatment are considerred to be same dict here
+                  propose_actions(sight, question_label, sample_list[id],
+                                  sample_list[id])))
+
+      print("waiting for all get outcome to finish.....")
+      diff_time_series = await asyncio.gather(*tasks)
+      print("all get outcome are finished.....")
+      print(f'Combine Series : {diff_time_series}')
+
+
+def get_decision_configuration_for_opt(sight, question_label, opt_obj,
+                                       question_config, optimizer_config):
+
+  # Extract attributes
+  action_attrs = decision_helper.config_to_attr(question_config, 'action')
+  state_attrs = decision_helper.config_to_attr(question_config, 'state')
+  outcome_attrs = decision_helper.config_to_attr(question_config, 'outcome')
+
+  sight.widget_decision_state[
+      'decision_episode_fn'] = decision_episode_fn.DecisionEpisodeFn(
+          state_attrs, action_attrs)
+  sight.widget_decision_state['num_decision_points'] = 0
+
+  # Set up the decision configuration
+  decision_configuration = sight_pb2.DecisionConfigurationStart(
+      optimizer_type=opt_obj.optimizer_type(),
+      num_trials=optimizer_config['num_questions'],
+      question_label=question_label)
+  decision_configuration.choice_config[sight.params.label].CopyFrom(
+      opt_obj.create_config())
+
+  decision_configuration.server_queue_batch_size = FLAGS.server_queue_batch_size or 1
+  decision_helper.attr_dict_to_proto(state_attrs,
+                                     decision_configuration.state_attrs)
+  decision_helper.attr_dict_to_proto(action_attrs,
+                                     decision_configuration.action_attrs)
+  decision_helper.attr_dict_to_proto(outcome_attrs,
+                                     decision_configuration.outcome_attrs)
+
+  # Enter and exit decision block
+  sight.enter_block(
+      'Decision Configuration',
+      sight_pb2.Object(block_start=sight_pb2.BlockStart(
+          sub_type=sight_pb2.BlockStart.ST_CONFIGURATION,
+          configuration=sight_pb2.ConfigurationStart(
+              sub_type=sight_pb2.ConfigurationStart.ST_DECISION_CONFIGURATION,
+              decision_configuration=decision_configuration,
+          ))))
+  sight.exit_block('Decision Configuration', sight_pb2.Object())
+  return decision_configuration
+
+
+def configure_decision(sight, question_label, question_config, optimizer_config,
+                       optimizer_type):
+  # Create optimizer and configure decision state
+  opt_obj = decision.setup_optimizer(sight, optimizer_type)
+  decision_configuration = get_decision_configuration_for_opt(
+      sight, question_label, opt_obj, question_config, optimizer_config)
+  return opt_obj, decision_configuration
+
+
+def start_worker_jobs(sight, optimizer_config, worker_configs, optimizer_type):
```

Comment:
Is this code that users will nee to write or is it something that would be placed within the sight library?